core/interfaces/i_grid.ts

```diff
@@ -0,0 +1,116 @@
+import {BlocklyOptions} from '../blockly_options.js';
+import {GridOptions} from '../options.js';
+import {Coordinate} from '../utils';
+import type {IRegistrable} from './i_registrable.js';
+
+export interface IGrid {
+  /**
+   * Sets the spacing between the centers of the grid lines.
+   *
+   * This does not trigger snapping to the newly spaced grid. If you want to
+   * snap blocks to the grid programmatically that needs to be triggered
+   * on individual top-level blocks. The next time a block is dragged and
+   * dropped it will snap to the grid if snapping to the grid is enabled.
+   */
+  setSpacing(spacing: number): void;
+
+  /**
+   * Get the spacing of the grid points (in px).
+   *
+   * @returns The spacing of the grid points.
+   */
+  getSpacing(): number;
+
+  /** Sets the length of the grid lines. */
+  setLength(length: number): void;
+
+  /** Get the length of the grid lines (in px). */
+  getLength(): number;
+
+  /**
+   * Sets whether blocks should snap to the grid or not.
+   *
+   * Setting this to true does not trigger snapping. If you want to snap blocks
+   * to the grid programmatically that needs to be triggered on individual
+   * top-level blocks. The next time a block is dragged and dropped it will
+   * snap to the grid.
+   */
+  setSnapToGrid(snap: boolean): void;
+
+  /**
+   * Whether blocks should snap to the grid.
+   *
+   * @returns True if blocks should snap, false otherwise.
+   */
+  shouldSnap(): boolean;
+
+  /**
+   * Get the ID of the pattern element, which should be randomized to avoid
+   * conflicts with other Blockly instances on the page.
+   *
+   * @returns The pattern ID.
+   * @internal
+   */
+  getPatternId(): string;
+
+  /**
+   * Update the grid with a new scale.
+   *
+   * @param scale The new workspace scale.
+   * @internal
+   */
+  update(scale: number): void;
```

Comment:
Let's make this `setScale()` and `getScale()` for consistency. The built-in `Grid` implementation can still call update() to adjust itself in response to changes in its configuration, but that doesn't need to be part of the interface (or require scale as an argument).